data_collection/gallica.py

```diff
@@ -0,0 +1,108 @@
+            description="Retrieve all the matching articles"
+        )
+        parser.add_argument("-r","--ref",help="reference of the searched newspaper")
+        parser.add_argument("-n","--newspaper",help="name of the searched newspaper")
+        parser.add_argument("-i","--issue",help="reference of the issue")
+        parser.add_argument("--start",type=int,help="year to start searching articles for (default 1900)")
+        parser.add_argument("--end",type=int,help="year up to which articles should be searched for (default 1950)")
+        parser.add_argument("--dir",help="""name of the directory to put issues in. 
+            if not set :
+                with --issue : print the issue to stdout
+                with --ref or --newspaper : print all references to stdout""")
+        args = parser.parse_args(stringArgs)
+        yearStart = 1900
+        yearEnd = 1950
+        if args.start :
+            yearStart = args.start
+        if args.end : 
+            yearEnd = args.end
+        if (int(args.ref!=None) + int(args.newspaper!=None) + int(args.issue!=None) != 1) :
+            print("exactly one of the --ref, --newspaper or --issue arguments must be given")
+        else :
+            if (int(args.ref!=None) + int(args.newspaper!=None) == 1):
+                if (args.ref) :
+                    refs = colart.get_issues_reference_by_newspaper_ref(args.ref,yearStart,yearEnd)
+                else :
+                    refs = colart.get_issues_reference_by_name(args.newspaper,yearStart,yearEnd)
+                if (args.dir) :
+                    p = Pool(8)
+                    tasks = []
+                    pbar=tqdm(total=len(refs))
+                    for ref in refs :
+                        tasks.append(p.apply_async(scrap_issue,(ref,args.dir),callback=lambda x:pbar.update()))
+                    for task in tasks:
+                        task.get()
+                else :
+                    print("\n".join(refs))
+            else :
+                issue = colart.get_issue_by_ref(args.issue)
+                if args.dir :
+                    extract.store_articles(issue,args.dir)
+                else :
+                    articles = extract.get_articles(issue["raw_text"])
+                    print("issue retrieved : ")
+                    print(f"newspaper : {issue['newspaper']}")
+                    print(f"date : {issue['date']}")
+                    print("\n")
+                    for article in articles :
+                        print(f"title : {article['title']}")
+                        print(article["text"])
+                        print("\n")
+
+#A helper function to paralellize scraping tasks
+def scrap_issue(*args):
+    ref = args[0]
+    directory = args[1]
+    try :
+        issue = colart.get_issue_by_ref(ref)
+        extract.store_articles(issue,directory)
+    except :
+        logging.info(f"invalid fromat for reference : {ref}")
+
+if __name__ == "__main__":
```

Comment:
Overall it's very good and very clear, a few minor updates can be done, and some new cool and good-practice-oriented libraries could be used !